yt_dlp/postprocessor/ffmpeg.py

```diff
@@ -1047,6 +1047,23 @@ def _ffmpeg_args_for_chapter(self, number, chapter, info):
             ['-ss', str(chapter['start_time']),
              '-t', str(chapter['end_time'] - chapter['start_time'])])
 
+    # FFmpeg adds metadata about all chapters from parent file to all split m4a files.
+    # This is incorrect since there must be only single chapter in each file after split.
+    # Such behavior confuses players who think multiple chapters present
+    def _set_out_opts(self, ext, chapter_title):
+        out_opts = [*self.stream_copy_opts()]
+        out_opts.extend(['-map_metadata', '0'])
+        # exclude chapters metadata but keep everything else
+        out_opts.extend(['-map_chapters', '-1'])
+        if not chapter_title:
+            return out_opts
+        # replace global title with chapter specific title in split files
+        if ext == 'm4a':
+            out_opts.extend(['-metadata', 'title={}'.format(chapter_title)])
+        if ext == 'opus':
+            out_opts.extend(['-metadata:s', 'title={}'.format(chapter_title)])
```

Comment:
Could you explain the `:s`? `FFmpegMetadataPP_get_metadata_opts` seems to work fine without it. Also, is it just title we want to change? For example, `track` can be chapter number, no?

Reply:
:s option is used to tell ffmpeg to set metadata(title in that case) on stream.
Ffmpeg produces different result when set metadata during split chapters mode and setting it on single file
It works with opus file when I set metadata on already split file.
Here is a gist with simple commands to reproduce: https://gist.github.com/danilovsergei/de27ec2de741f0f60d7433082ab534f8
It also explains why FFmpegMetadataPP works , because ffmpeg is not doing split chapters together with setting metadata in one command

I tested on various youtube videos and it seems to be consistent behavior with opus format and ffmpeg.

And sure, we can set all other available information in the chapter object. I added code

Reply:
I knew what :s is, but why it's needed in this case wasn't obvious. Could u add a comment to the code, so future me dont accidentally get rid of it

> And sure, we can set all other available information in the chapter object. I added code

Forgot to push?

Reply:
Added code.
I didn't expect vscode will make my comment public right after pressing 'Add comment' without event sending code.

I did more digging by doing ffprobe on opus files.
Issue is more broad than just 'split mode' and related to 
ffmpeg is using the stream to save opus metadata , such as artist/title

Opus files downloaded without --add-metadata do not have a stream
https://gist.github.com/danilovsergei/c4e3e92c200aef88e217793d3d882f58

While opus files downloaded with --add-metadata have a stream
https://gist.github.com/danilovsergei/288784f8512e05f3489d8b0277820bf3

--add-metadata triggers FFmpegMetadataPP to set title/artist etc.
New stream is created inside opus file after that.

All further metadata modifications on opus file will require -metadata:flag

I added a comment to _set_metadata_arg function